theme/redbox.js

```diff
@@ -36,3 +36,69 @@
     }
   });
 })();
+
+function setCodeToPlayground() {
+  const code = JSON.parse(localStorage.getItem(window.location.href));
+  console.log(code);
+  if (code) {
+    const playground = document.getElementsByClassName("ace_text-layer")[0];
+    while (playground.firstElementChild) {
+      console.log(playground.firstElementChild.innerText.replace(/\s+/g, " "));
+      playground.removeChild(playground.firstElementChild);
+    }
+    console.log(playground, "after removal");
+    for (let i = 0; i < code.length; i++) {
+      let parentDiv = code[i][0];
+      let spanChild = code[i][1];
+      let div = document.createElement(parentDiv.tag);
+      div.style.height = "17.5938px";
+      div.style.top = `${17.5938 * i}px`;
+      for (let cls in parentDiv.classes) {
+        div.classList.add(parentDiv.classes[cls]);
+      }
+      for (let j = 0; j < spanChild.length; j++) {
+        //console.log(spanChild[j].styles,typeof(spanChild[j].styles))
+        let span = document.createElement(spanChild[j].tag);
+        //span.classList = spanChild[j].classes
+        for (let cls in spanChild[j].classes) {
+          span.classList.add(spanChild[j].classes[cls]);
+        }
+        span.innerText = spanChild[j].text;
+        div.insertBefore(span, div.lastChild);
+      }
+      playground.insertBefore(div, playground.lastChild);
+    }
+  }
```

Comment:
Use the API of the editor component rather than poking into its portion of the DOM:
```js
editors[0].setValue("fn main(){\n\n}", -1);
```
or similar.